src/test/java/FreeTimeServletTest.java

```diff
@@ -0,0 +1,211 @@
+    Mockito.when(calendarClient.getCalendarList()).thenReturn(ONE_CALENDAR);
+    Mockito.when(calendarClient.getUpcomingEvents(PRIMARY, CURRENT_TIME, END_TIME))
+        .thenReturn(NO_EVENT);
+    Mockito.when(calendarClient.getCurrentTime()).thenReturn(CURRENT_TIME);
+    Mockito.when(gmailClient.getWordCount(7)).thenReturn(0);
+    PlanMailResponse actual = getServletResponse();
+    Assert.assertEquals(0, actual.getWordCount());
+    Assert.assertEquals(50, actual.getAverageReadingSpeed());
+    Assert.assertEquals(0, actual.getMinutesToRead());
+    Assert.assertTrue(actual.getPotentialEventTimes().isEmpty());
+  }
+
+  @Test
+  public void onePotentialEvent() throws Exception {
+    // Test case where there are unread emails that will take 10 minutes to read
+    // We should get 1 proposed event that takes 10 minutes
+    Mockito.when(calendarClient.getCalendarList()).thenReturn(ONE_CALENDAR);
+    Mockito.when(calendarClient.getUpcomingEvents(PRIMARY, CURRENT_TIME, END_TIME))
+        .thenReturn(EVENT_ONE);
+    Mockito.when(calendarClient.getCurrentTime()).thenReturn(CURRENT_TIME);
+    Mockito.when(gmailClient.getWordCount(7)).thenReturn(500);
+    PlanMailResponse actual = getServletResponse();
+    List<Date> startHour = Arrays.asList(new Date(2020 - OFFSET_YEAR, 4, 19, 10, 0));
+    List<Date> endHour = Arrays.asList(new Date(2020 - OFFSET_YEAR, 4, 19, 10, 10));
+    Assert.assertEquals(500, actual.getWordCount());
+    Assert.assertEquals(50, actual.getAverageReadingSpeed());
+    Assert.assertEquals(10, actual.getMinutesToRead());
+    Assert.assertEquals(startHour.size(), actual.getPotentialEventTimes().size());
+    for (int index = 0; index < startHour.size(); index++) {
+      Assert.assertTrue(
+          startHour.get(index).equals(actual.getPotentialEventTimes().get(index).getStart()));
+      Assert.assertTrue(
+          endHour.get(index).equals(actual.getPotentialEventTimes().get(index).getEnd()));
+    }
+  }
+
+  @Test
+  public void alreadyCreatedEvent() throws Exception {
+    // Test case where there are unread emails that will take 10 minutes to read
+    // But an event (event 2) has already been created to go through them
+    // We should get no proposed potential event
+    Mockito.when(calendarClient.getCalendarList()).thenReturn(TWO_CALENDARS);
+    Mockito.when(calendarClient.getUpcomingEvents(PRIMARY, CURRENT_TIME, END_TIME))
+        .thenReturn(EVENT_ONE);
+    Mockito.when(calendarClient.getUpcomingEvents(SECONDARY, CURRENT_TIME, END_TIME))
+        .thenReturn(EVENT_TWO);
+    Mockito.when(calendarClient.getCurrentTime()).thenReturn(CURRENT_TIME);
+    Mockito.when(gmailClient.getWordCount(7)).thenReturn(500);
+    PlanMailResponse actual = getServletResponse();
+    Assert.assertEquals(500, actual.getWordCount());
+    Assert.assertEquals(50, actual.getAverageReadingSpeed());
+    Assert.assertEquals(10, actual.getMinutesToRead());
+    Assert.assertTrue(actual.getPotentialEventTimes().isEmpty());
```

Comment:
If you override `equals` in `PlanMailResponse`, you can do `Assert.assertEquals(response1, response2)` but it's up to you, it's the last day haha

Reply:
I agree, it's much cleaner. `Resolved`

Reply:
Thanks! `RESOLVED`